xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp

```diff
@@ -162,16 +162,16 @@ bool CDVDDemuxClient::ParsePacket(DemuxPacket* pkt)
     avpkt->size = pkt->iSize;
     avpkt->dts = avpkt->pts = AV_NOPTS_VALUE;
 
-    AVCodecParameters* codecPar = avcodec_parameters_alloc();
-    int ret = avcodec_parameters_from_context(codecPar, stream->m_context);
+    auto codecPar = std::unique_ptr<AVCodecParameters, void (*)(AVCodecParameters*)>(
+        avcodec_parameters_alloc(), [](AVCodecParameters* p) { avcodec_parameters_free(&p); });
```

Comment:
a style suggestion i have (not sure if `constexpr` makes sense here, but it builds)

```cpp
  constexpr auto deleter = [](AVCodecParameters* p) { avcodec_parameters_free(&p); };

  auto codecPar = std::unique_ptr<AVCodecParameters, decltype(deleter)>(
    avcodec_parameters_alloc(), deleter);
```

Reply:
I like it :) Thanks
Modified accordingly